src/rust/src/transaction_ffi.rs

```diff
@@ -85,6 +123,54 @@ pub extern "C" fn zcash_transaction_precomputed_init(
             ptr::null_mut()
         }
         _ => {
+            let all_prev_outputs =
+                unsafe { slice::from_raw_parts(all_prev_outputs, all_prev_outputs_len) };
+
+            // TODO: Align this with the libzcash_script API for efficiency.
+            let mut cursor = Cursor::new(all_prev_outputs);
+            let all_prev_outputs = match Vector::read(&mut cursor, transparent::TxOut::read) {
+                Err(e) => {
+                    error!("Invalid all_prev_outputs field: {}", e);
+                    return ptr::null_mut();
+                }
+                Ok(_) if (cursor.position() as usize) != all_prev_outputs.len() => {
+                    error!("all_prev_outputs had trailing data");
+                    return ptr::null_mut();
+                }
+                Ok(res) => res,
+            };
+
+            let transparent_bundle =
+                tx.transparent_bundle()
+                    .as_ref()
+                    .map(|b| transparent::Bundle {
+                        // we have to do this map/clone to make the types line up, since the
+                        // Authorization::ScriptSig type is bound to transparent::Authorized, and we need
+                        // it to be bound to TestTransparentAuth.
```

Comment:
```suggestion
                        // it to be bound to TransparentAuth.
```

The way I had anticipated doing this was actually adding a method `Transaction::read_without_authorization` that takes the `Vec<TxOut>` of previous outputs as an argument. produces a `Transaction<Unauthorized>`. The way you're doing this here is less invasive, but I think it's a little bit fragile; having the only API our Rust library provides be "read this transaction with fake authorizing data, then you go and manually replace those pieces with the right things" seems error-prone. On the plus side, the trait bound on signatures means that you're mostly protected from doing the wrong thing, but I'm worried that the fact one needs to use the API in this fashion is not at all obvious.